src/state/user/updater.tsx

```diff
@@ -1,10 +1,24 @@
+import { L2_CHAIN_IDS } from 'constants/chains'
+import { DEFAULT_DEADLINE_FROM_NOW, L2_DEADLINE_FROM_NOW } from 'constants/misc'
+import { useActiveWeb3React } from 'hooks/web3'
 import { useEffect } from 'react'
 import { useAppDispatch } from 'state/hooks'
 
 import { updateMatchesDarkMode } from './actions'
+import { useUserTransactionTTL } from './hooks'
 
 export default function Updater(): null {
   const dispatch = useAppDispatch()
+  const { chainId } = useActiveWeb3React()
+  const [, setTTL] = useUserTransactionTTL()
+
+  // keep deadline appropriate to chainId
+  useEffect(() => {
+    if (!chainId) {
+      return
+    }
+    setTTL(L2_CHAIN_IDS.includes(chainId) ? L2_DEADLINE_FROM_NOW : DEFAULT_DEADLINE_FROM_NOW)
```

Comment:
should not use useEffect like this. the result is if the user switches back to the main chain they lost their settings. instead we should just update useTransactionDeadline to return a diff deadline and the deadline input on optimism